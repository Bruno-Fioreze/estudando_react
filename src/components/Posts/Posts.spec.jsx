import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    { id: 1, title: 'title 1', body: 'body1', cover: 'im1g.png' },
    { id: 2, title: 'title 2', body: 'body2', cover: 'im2g.png' },
    { id: 3, title: 'title 3', body: 'body3', cover: 'im4g.png' }
  ]
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/title/i)).toHaveLength(3);
  });
  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
