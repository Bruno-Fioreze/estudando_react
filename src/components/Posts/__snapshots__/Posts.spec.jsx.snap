// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="im1g.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 1
      </h2>
      <p>
        body1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="im2g.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 2
      </h2>
      <p>
        body2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="im4g.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 3
      </h2>
      <p>
        body3
      </p>
    </div>
  </div>
</div>
`;
